tests/test_model.py

```diff
@@ -49,3 +54,104 @@ def model(cfg):
     testval_mode = model.test_point(sample=False)
     assert testval_mode['normal'] == 0.
     assert testval_mode['normal'] != testval_random['normal']
+
+
+def test_variables():
+    model = pm.Model()
+
+    @model.define
+    def simple(cfg):
+        ed.Normal(0., 1., name='normal')
+
+    assert len(model.variables) == 1
+    assert len(model.unobserved) == 1
+    assert "normal" in model.variables
+
+
+def test_model_target_log_prob_fn():
+    model = pm.Model()
+
+    @model.define
+    def simple(cfg):
+        ed.Normal(0., 1., name='normal')
+
+    model.target_log_prob_fn()
+
+
+def test_model_observe():
+
+    model = pm.Model()
+
+    @model.define
+    def simple(cfg):
+        ed.Normal(0., 1., name='normal')
+
+    model.observe(normal=1)
+
+    assert len(model.observed) == 1
+    assert not model.unobserved
+
+
+def test_model_reset():
+    model = pm.Model()
+
+    @model.define
+    def simple(cfg):
+        ed.Normal(0., 1., name='normal')
+
+    model.observe(normal=1)
+
+    assert len(model.observed) == 1
+    assert not model.unobserved
+
+    model.reset()
```

Comment:
We decided to meke a copy of model each time state changes

Reply:
this one is not critical though, refactoring interceptor usage is what is really needed to finish this PR (https://github.com/pymc-devs/pymc4/pull/9#discussion-diff-200027548R87)